### PR TITLE
feat(workflow): Add GroupInbox support to post_process_group

### PR DESCRIPTION
**Pull request by iProgramStuff**
As part of the work towards supporting the new issues inbox, we're now using the GroupInbox model to track groups as appropriate. We currently track 3 inbox states (new, unignored, regressed), and groups are currently added to the inbox in during post processing. 

Groups cannot currently be removed from the inbox. This will be in a future PR.

### CODE REVIEW

File: src/sentry/models/groupinbox.py [hunk 1]
```diff
@@ -29,3 +29,16 @@ class GroupInbox(Model):
...
         db_table = "sentry_groupinbox"
+
+
+def add_group_to_inbox(group, reason, reason_details):
+    if not GroupInbox.objects.filter(group=group).exists():
+        return GroupInbox.objects.create(
+            group=group, reason=reason.value, reason_details=reason_details
+        )
+    else:
+        return None
```
Comment (wedamija) [inline src/sentry/models/groupinbox.py:40]: Could potentially just use `get_or_create` here.

File: src/sentry/models/groupinbox.py [hunk 2]
```diff
@@ -29,3 +29,16 @@ class GroupInbox(Model):
...
+
+
+def remove_group_from_inbox(group):
+    GroupInbox.objects.filter(group=group).delete()
```
Comment (wedamija) [inline src/sentry/models/groupinbox.py:44]: It's fine to do it this way, just be aware it won't raise any Django signals.

File: src/sentry/tasks/post_process.py [hunk 3]
```diff
@@ -279,6 +295,14 @@ def process_snoozes(group):
         return False
 
     if not snooze.is_valid(group, test_rates=True):
+        snooze_details = {
+            "until": snooze.until,
+            "count": snooze.count,
+            "window": snooze.window,
+            "user_count": snooze.user_count,
+            "user_window": snooze.user_window,
+        }
```
Comment (wedamija) [inline src/sentry/tasks/post_process.py:304]: Still want to see a schema for this.

File: src/sentry/models/groupinbox.py (hunk 1 above)
Comment (iProgramStuff) [reply src/sentry/models/groupinbox.py:40]: Hmm true...was concerned about the possibility of this being called with a different reason when it's already in the inbox

File: src/sentry/tasks/post_process.py (hunk 3 above)
Comment (iProgramStuff) [reply src/sentry/tasks/post_process.py:304]: Schema: Coming to a PR near you before internal

File: src/sentry/models/groupinbox.py [hunk 4]
```diff
@@ -29,3 +29,20 @@ class GroupInbox(Model):
...
+            group=group, reason=reason.value, reason_details=reason_details
+        )
+    else:
+        return None
+
+
+def remove_group_from_inbox(group):
+    try:
+        group = GroupInbox.objects.get(group=group).delete()
+        group.delete()
```
Comment (wedamija) [inline src/sentry/models/groupinbox.py:46]: I think it might be worth testing these both directly as well as via `post_process`

**Review by wedamija — APPROVED** [review]

File: src/sentry/models/groupinbox.py [hunk 5]
```diff
@@ -29,3 +29,18 @@ class GroupInbox(Model):
...
+    gi, created = GroupInbox.objects.get_or_create(
+        group=group, defaults={"reason": reason.value, "reason_details": reason_details}
+    )
+    return gi
+
+
+def remove_group_from_inbox(group):
+    try:
+        group = GroupInbox.objects.get(group=group)
+        group.delete()
```
Comment (markstory) [inline src/sentry/models/groupinbox.py:44]: Its confusing that you reuse the name `group` for the inbox item when it was the Group.

File: tests/sentry/tasks/post_process/tests.py [hunk 6]
```diff
@@ -573,3 +585,45 @@ def test_processes_resource_change_task_not_called_without_error_created(self, d
...
+    def test_group_inbox_regression(self, mock_processor):
+        from sentry.models import GroupInbox, GroupInboxReason
+
+        event = self.store_event(data={"message": "testing"}, project_id=self.project.id)
+        cache_key = write_event_to_cache(event)
+
+        group = event.group
+
+        post_process_group(
+            event=None,
```
Comment (markstory) [inline tests/sentry/tasks/post_process/tests.py:599]: This parameter was removed.

Comment (iProgramStuff) [reply tests/sentry/tasks/post_process/tests.py]: Oh, thanks! Merged and removed

File: src/sentry/models/groupinbox.py (hunk 5 above)
Comment (iProgramStuff) [reply src/sentry/models/groupinbox.py]: I agree, changed this.

**Pull request merged**